src/lib/Microsoft.Health.Fhir.Ingest/Service/NormalizationStreamingEventConsumerService.cs

```diff
@@ -0,0 +1,142 @@
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class NormalizationStreamingEventConsumerService : StreamingEventConsumerService
+    {
+        private readonly IContentTemplate _contentTemplate;
+        private readonly IConverter<IEventMessage, JObject> _converter;
+        private readonly IExceptionTelemetryProcessor _exceptionTelemetryProcessor;
+        private readonly IEnumerableAsyncCollector<IMeasurement> _collector;
+
+        public NormalizationStreamingEventConsumerService(
+            EventBatchingOptions options,
+            ICheckpointClient checkpointClient,
+            ITelemetryLogger log,
+            IEventProcessingMetricMeters eventProcessingMetricMeters,
+            IConverter<IEventMessage, JObject> converter,
+            IContentTemplate contentTemplate,
+            IExceptionTelemetryProcessor exceptionTelemetryProcessor,
+            IEnumerableAsyncCollector<IMeasurement> collector)
+            : base(options, checkpointClient, log, eventProcessingMetricMeters)
+        {
+            _contentTemplate = EnsureArg.IsNotNull(contentTemplate, nameof(contentTemplate));
+            _converter = EnsureArg.IsNotNull(converter, nameof(converter));
+            _exceptionTelemetryProcessor = EnsureArg.IsNotNull(exceptionTelemetryProcessor, nameof(exceptionTelemetryProcessor));
+            _collector = EnsureArg.IsNotNull(collector, nameof(collector));
+        }
+
+        /// <summary>
+        /// Reusable list to collect projected measurements
+        /// </summary>
+        private List<IMeasurement> NormalizationBatch { get; } = new List<IMeasurement>(50);
+
+        protected override async Task ConsumeEventImpl(IEventMessage eventArg)
+        {
+            try
+            {
+                RecordLatencyMetrics(eventArg);
+
+                var token = _converter.Convert(eventArg);
+
+                await NormalizeMessage(token, eventArg);
+            }
+            catch (Exception ex)
+            {
+                if (!_exceptionTelemetryProcessor.HandleException(ex, Logger))
+                {
+                    throw; // Immediately throw originalexception if it is not handled
+                }
+            }
+        }
+
+        private async Task NormalizeMessage(JObject token, IEventMessage eventArg)
+        {
+            Stopwatch sw = Stopwatch.StartNew();
+            foreach (var measurement in _contentTemplate.GetMeasurements(token))
+            {
+                try
+                {
+                    measurement.IngestionTimeUtc = eventArg.EnqueuedTime.UtcDateTime;
+                    NormalizationBatch.Add(measurement);
+                }
+                catch (Exception ex)
+                {
+                    // Translate all Normalization Mapping exceptions into a common type for easy identification.
+                    throw new NormalizationDataMappingException(ex);
+                }
+            }
+
+            sw.Stop();

```

Comment:
This will record the time to collect all of the projections. It would also be nice to have the time it takes to produce a single projection. But we can add that telemetry in later.

Reply:
Good point, I was thinking we had a NormalizationStreamingEventConsumerService per partition but that isn't the case.  I switched to a new instance local in the function for now.